src/Core/Scheduler/Scheduler.js

```diff
@@ -162,8 +161,14 @@ Scheduler.prototype.execute = function execute(command) {
     return command.promise;
 };
 
-
 Scheduler.prototype.addProtocolProvider = function addProtocolProvider(protocol, provider) {
+    if (typeof (provider.executeCommand) !== 'function') {
+        throw new Error(`Can't add provider for ${protocol}: missing a executeCommand function.`);
+    }
+    if (typeof (provider.preprocessDataLayer) !== 'function') {
+        throw new Error(`Can't add provider for ${protocol}: missing a preprocessDataLayer function.`);
+    }
+
```

Comment:
Nice! That's the first step to *really* allow people to add their own providers (I mean, it was already possible, but you needed to know the internals of iTowns). Second step would be to document it :-)

Reply:
What do you have in mind for documentation ? Something like iTowns/itowns.github.io#15 ? I can't see where else we could add it.

Reply:
Maybe if we document the `Scheduler` in fact

Reply:
Using http://usejsdoc.org/tags-interface.html#virtual-comments in the Scheduler makes sense.
But maybe you should finish your refactoring commits before adding documentation...

Reply:
Yes, will do in another PR.